examples/06-custom-effects/README.md

```diff
@@ -1,24 +1,77 @@
+![](./header.png)
+
 # 06 Custom Effects
 
-We haven't quite got round to documenting this example yet. If you know a little
-bit about Lustre or Elm and want to help out, we'd love to have your help! Please
-[open an issue](https://github.com/lustre-labs/lustre/issues/new) if you have any
-ideas or reach out to @hayleigh.dev on the [Gleam discord](https://discord.gg/Fm8Pwmy).
+In the last example, we showed how to use effects provided by `lustre_http`. In
+this example we'll see how to create effects of our own using Lustre's
+[`effect.from`](https://hexdocs.pm/lustre/4.0.0-rc1/lustre/effect.html#from)
+method.
+
+Since we use Lustre effects to communicate with _external_ systems (like the
+browser or the Erlang VM) you'll find creating custom effects usually involves
+Gleam's [external
+functions](https://tour.gleam.run/everything/#advanced-features-externals). So
+be sure to read up on that!
+
+> Gleam externals are part of its "foreign function interface", which is why
+> you'll typically see files with `ffi` in the name - like
+> [`app.ffi.mjs`](./src/app.ffi.mjs).
+
+## Accessing Browser Storage
+
+In this example, the external system we want to interact with is the browser's
+local storage. This way, we can write a message into the text input and it will
+still be there when we refresh the page. Handy!
+
+The `view`, `update` and `init` functions should look pretty familiar by now, so
+let's focus on the functions that generate our custom effects.
+
+```rust
+fn read_localstorage(key: String) -> Effect(Msg) {
+    effect.from(fn(dispatch) {
+	do_read_localstorage(key) |> CacheUpdatedMessage |> dispatch
+    })
+}
+```
+
+We use `effect.from` by passing it a custom function that describes the effect
+we want the Lustre runtime to perform. Our custom function will receive a
+`dispatch` function as its only parameter - we can use that to send new messages
+back to our `update` function if we want to.
+
+In this case, we read from local storage, pipe its value into the
+`CacheUpdatedMessage` constructor, and pipe that to the `dispatch` function so
+our `update` messsage can handle it.
+
+```rust
+fn write_localstorage(key: String, value: String) -> Effect(msg) {
+    effect.from(fn(_) {
+        do_write_localstorage(key, value)
+    })
+}
+```
+
+Here, our effect is simpler. We tell the gleam compiler we don't need to use the
+`dispatch` function by replacing it with a [discard
+pattern](https://tour.gleam.run/everything/#basics-discard-patterns). Then we
+write to local storage, and no more work needs to be done.
```

Comment:
Maybe we could add another paragraph explaining why you'd still want to use an effect here even though we're not making use of the `dispatch`. Moving side effects out our `update` function makes it easier to test and easier to port to a [server component](https://hexdocs.pm/lustre/lustre/server_component.html) in the future.

Reply:
Updated! Does this work?

Reply:
Stunning, thank you!